src/pkg/packager/prepare.go

```diff
@@ -33,8 +33,41 @@ import (
 // imageMap is a map of image/boolean pairs.
 type imageMap map[string]bool
 
+// FindImagesWithPackage iterates over a ZarfPackage object and attempts to parse images.
+func (p *Packager) FindImagesWithPackage() error {
+	message.Debug("Finding images for ZarfPackage")
+	if p.layout.IsLegacyLayout() {
+		message.Debug("p.layout.IsLegacyLayout = true")
+	}
+	p.arch = config.GetArch(p.cfg.Pkg.Metadata.Architecture, p.cfg.Pkg.Build.Architecture)
+
+	images, err := p.findImages()
+	if err != nil {
+		return err
+	}
+
+	for i := range p.cfg.Pkg.Components {
+		p.cfg.Pkg.Components[i].Images = images[p.cfg.Pkg.Components[i].Name]
+	}
+
+	return nil
+}
+
 // FindImages iterates over a Zarf.yaml and attempts to parse any images.
 func (p *Packager) FindImages() (imgMap map[string][]string, err error) {
+	if err := os.Chdir(p.cfg.CreateOpts.BaseDir); err != nil {
```

Comment:
I suggest refactoring the FindImages function to avoid changing the process's working directory. By constructing and using absolute paths directly, we can enhance the robustness and predictability of our file operations. This approach also helps avoid unintended side effects on the application's global state, which is particularly beneficial in a multi-threaded environment. Here's a proposed change:

``` golang
// FindImages iterates over a Zarf.yaml and attempts to parse any images.
func (p *Packager) FindImages() (imgMap map[string][]string, err error) {
    // Validate BaseDir before proceeding
    if err := validateBaseDir(p.cfg.CreateOpts.BaseDir); err != nil {
        return nil, err
    }

    // Construct the absolute path to the Zarf.yaml file
    zarfYAMLPath := filepath.Join(p.cfg.CreateOpts.BaseDir, layout.ZarfYAML)

    // Use the absolute path to read the Zarf.yaml file
    if err = p.readZarfYAML(zarfYAMLPath); err != nil {
        return nil, fmt.Errorf("unable to read the zarf.yaml file at '%s': %w", zarfYAMLPath, err)
    }

    // Ensure that p.findImages() and any functions it calls are modified to work with absolute paths
    return p.findImages()
}

// validateBaseDir validates the BaseDir path for security and correctness.
func validateBaseDir(baseDir string) error {
    // Ensure the path is absolute
    if !filepath.IsAbs(baseDir) {
        return fmt.Errorf("BaseDir must be an absolute path")
    }

    // Canonicalize the path
    cleanPath := filepath.Clean(baseDir)

    // Check for existence and directory type
    info, err := os.Stat(cleanPath)
    if err != nil {
        return fmt.Errorf("BaseDir does not exist: %w", err)
    }
    if !info.IsDir() {
        return fmt.Errorf("BaseDir is not a directory")
    }

    // Optionally, check for read/write permissions as needed

    return nil
}
```

On the flip side, if all those security checks aren't required, here is a simple one.
``` golang
// FindImages iterates over a Zarf.yaml and attempts to parse any images.
func (p *Packager) FindImages() (imgMap map[string][]string, err error) {
    // Construct the absolute path to the Zarf.yaml file
    zarfYAMLPath := filepath.Join(p.cfg.CreateOpts.BaseDir, layout.ZarfYAML)

    // Use the absolute path to read the Zarf.yaml file
    if err = p.readZarfYAML(zarfYAMLPath); err != nil {
        return nil, fmt.Errorf("unable to read the zarf.yaml file at '%s': %w", zarfYAMLPath, err)
    }

    // Ensure that p.findImages() and any functions it calls are modified to work with absolute paths
    return p.findImages()
}
```
_not compiled or tested_

Reply:
The `FindImages` is prone to directory path traversal attacks, which is also one reason for the above suggestion.

Reply:
Great Feedback! The refactoring I've done on FindImages was intentionally narrow in scope. I have created https://github.com/defenseunicorns/zarf/issues/2336 to address your comments.